### ISSUE TITLE
std/http/file_server exits process and prints own usage when imported and -h/--help in program args

### ISSUE THREAD

**Issue by ejones** [denoland/deno#5973]
**To reproduce:**

```
// file_server_test.ts

import { serve } from "https://deno.land/std@0.53.0/http/server.ts";
import { serveFile } from "https://deno.land/std@0.53.0/http/file_server.ts";
import { parse } from "https://deno.land/std@0.53.0/flags/mod.ts";

const args = parse(Deno.args);
if (args.h) {
  console.log('Usage: test');
  Deno.exit();
}

for await (const req of serve({ port: 8080 })) {
  serveFile(req, 'foo');
}
```
```
$ deno run --allow-read file_server_test.ts -h
```

**Actual:** prints out "Deno File Server" usage, does not print out script's own usage
```
$ deno --version
deno 1.0.3
v8 8.4.300
typescript 3.9.2
$ deno run --allow-read file_server_test.ts -h
...
Deno File Server
  Serves a local directory in HTTP.
...
```

**Expected:** when imported as a module, `file_server` should not make use of the program args, exit the process, or print anything to the console.

**Comment by ry** [comment]
Yeah - ideally file_server can be imported and used in other servers

**Comment by marchiore** [comment]
I'm looking at the code trying to understand the problem and find a solution to it and i notice that `file_server` uses Deno args thats the same of script `const args = parse(Deno.args);`

https://github.com/denoland/deno/blob/64bd2768f7794408a9e8b7ba14edc990f3e2d0de/std/http/file_server.ts#L9

Is there a way to check if file_server is imported inside a script? if we can have a flag for this we can ignore de args?

i'm looking for a good first issue, if this is not very complex I would like to work on it

**Comment by ejones** [comment]
@marchiore `import.meta.main` indicates if a script was invoked directly rather than being imported. `file_server` is already using this to decide whether to actually serve files. I imagine the fix will involve moving the arg parsing in some way inside the `import.meta.main` check or the `main()` function.

**Comment by marchiore** [comment]
thank you @ejones  it made things clearer for me. 

I will work on it and send a PR
